images/sds-replicated-volume-controller/src/pkg/controller/replicated_storage_class.go

```diff
@@ -578,56 +674,111 @@ func recreateStorageClassIfNeeded(ctx context.Context, cl client.Client, log log
 
 func GetNewStorageClass(replicatedSC *srv.ReplicatedStorageClass, virtualizationEnabled bool) *storagev1.StorageClass {
 	newSC := GenerateStorageClassFromReplicatedStorageClass(replicatedSC)
-	newSC.Labels = map[string]string{ManagedLabelKey: ManagedLabelValue}
 	if replicatedSC.Spec.VolumeAccess == VolumeAccessLocal && virtualizationEnabled {
-		newSC.Annotations = map[string]string{StorageClassVirtualizationAnnotationKey: StorageClassVirtualizationAnnotationValue}
+		if newSC.Annotations == nil {
+			newSC.Annotations = make(map[string]string, 1)
+		}
+		newSC.Annotations[StorageClassVirtualizationAnnotationKey] = StorageClassVirtualizationAnnotationValue
 	}
 	return newSC
 }
 
-func GetUpdatedStorageClass(replicatedSC *srv.ReplicatedStorageClass, oldSC *storagev1.StorageClass, virtualizationEnabled bool) *storagev1.StorageClass {
-	newSC := GenerateStorageClassFromReplicatedStorageClass(replicatedSC)
-
-	newSC.Labels = make(map[string]string, len(oldSC.Labels))
-	for k, v := range oldSC.Labels {
-		newSC.Labels[k] = v
+func DoUpdateStorageClass(
+	newSC *storagev1.StorageClass,
+	oldSC *storagev1.StorageClass,
+) {
+	// Copy Labels from oldSC to newSC if they do not exist in newSC
+	if len(oldSC.Labels) > 0 {
+		if newSC.Labels == nil {
+			newSC.Labels = maps.Clone(oldSC.Labels)
+		} else {
+			updateMap(newSC.Labels, oldSC.Labels, false)
+		}
 	}
-	newSC.Labels[ManagedLabelKey] = ManagedLabelValue
 
-	newSC.Annotations = make(map[string]string, len(oldSC.Annotations))
-	for k, v := range oldSC.Annotations {
-		newSC.Annotations[k] = v
-	}
+	copyAnotations := maps.Clone(oldSC.Annotations)
+	delete(copyAnotations, StorageClassVirtualizationAnnotationKey)
```

Comment:
We must delete the annotation only if virtualizationEnabled is false and VolumeAccess is not set to Local.